src/routes/api/tasks/index.ts

```diff
@@ -1,31 +1,50 @@
-import {
-  FastifyPluginAsyncTypebox,
-  Type
-} from '@fastify/type-provider-typebox'
-import {
-  TaskSchema,
-  Task,
-  CreateTaskSchema,
-  UpdateTaskSchema,
-  TaskStatus
-} from '../../../schemas/tasks.js'
-import { FastifyReply } from 'fastify'
+import { FastifyPluginAsyncTypebox, Type } from '@fastify/type-provider-typebox'
+import { TaskSchema, Task, CreateTaskSchema, UpdateTaskSchema, TaskStatusEnum, QueryTaskPaginationSchema, TaskPaginationResultSchema } from '../../../schemas/tasks.js'
 
 const plugin: FastifyPluginAsyncTypebox = async (fastify) => {
   fastify.get(
     '/',
     {
       schema: {
+        querystring: QueryTaskPaginationSchema,
         response: {
-          200: Type.Array(TaskSchema)
+          200: TaskPaginationResultSchema
         },
         tags: ['Tasks']
       }
     },
-    async function () {
-      const tasks = await fastify.repository.findMany<Task>('tasks')
+    async function (request) {
+      const q = request.query
 
-      return tasks
+      const offset = (q.page - 1) * q.limit
+
+      const baseQuery = fastify.knex('tasks')
+
+      if (q.author_id !== undefined) {
+        baseQuery.where({ author_id: q.author_id })
+      }
+
+      if (q.assigned_user_id !== undefined) {
+        baseQuery.where({ assigned_user_id: q.assigned_user_id })
+      }
+
+      if (q.status !== undefined) {
+        baseQuery.where({ status: q.status })
+      }
+
+      const tasksQuery = baseQuery.clone()
+        .select('*')
+        .limit(q.limit).offset(offset).orderBy('created_at', q.order)
+
+      const [tasks, [{ count }]] = await Promise.all([
+        tasksQuery,
+        baseQuery.clone().count({ count: '*' })
+      ])
```

Comment:
This is a bad practice and it's inefficient. Maybe use some custom SQL?

Reply:
ok

Reply:
Maybe, 
```sql
SELECT *, COUNT(*) OVER() AS rowNum
FROM tasks
WHER ...
```
Not sure how it goes into knex.

Reply:
The problem is that it put the row (`total`) in every task:
```ts
 {
    id: 95,
    name: 'Task 3',
    author_id: 21,
    assigned_user_id: null,
    status: 'completed',
    created_at: 2024-10-18T06:02:07.000Z,
    updated_at: 2024-10-18T06:02:07.000Z,
    total: 89
  },
  {
    id: 94,
    name: 'Task 2',
    author_id: 20,
    assigned_user_id: 21,
    status: 'in-progress',
    created_at: 2024-10-18T06:02:07.000Z,
    updated_at: 2024-10-18T06:02:07.000Z,
    total: 89
  }
// ...
```

Reply:
Can you take a look at [this commit](https://github.com/fastify/demo/pull/33/commits/da0fb0da7a710e079d0a86a2ae3eeeb5d90f919e) plz?